App/Sagas/WaitFor.ts

```diff
@@ -1,14 +1,14 @@
-import { select, take } from 'redux-saga/effects'
+import { select, take, SelectEffect } from 'redux-saga/effects'
 import { RootState } from '../Redux/Types'
 
-export function * waitFor (selector: (state: RootState) => boolean) {
-  if (yield select(selector)) {
-    return
+export function * waitFor (selectEffect: SelectEffect) {
```

Comment:
This is a reusable effect I created that makes it easy to block a saga until some state you are interested in exists in the redux store.

Reply:
OS it